lib/Doctrine/Driver/PDOConnection.php

```diff
@@ -39,15 +40,21 @@ public function exec($statement) {
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
   public function getServerVersion() {
-    return $this->connection->getAttribute(\PDO::ATTR_SERVER_VERSION);
+    $version = $this->connection->getAttribute(\PDO::ATTR_SERVER_VERSION);
+    return is_scalar($version) ? (string)$version : '';
   }
 
   public function prepare($prepareString, $driverOptions = []) {
     try {
-      return $this->createStatement(
-        $this->connection->prepare($prepareString, $driverOptions)
-      );
+      $preparedStatement = $this->connection->prepare($prepareString, $driverOptions);
+      if ($preparedStatement === false) {
+        throw new RuntimeException('Unable to prepare PDOStatement for ' . $prepareString);
```

Comment:
Is this safe enough? Can we throw the prepared string? What if we leak any private data here? Can we?